profiler/options.go

```diff
@@ -111,6 +144,20 @@ func defaultConfig() *config {
 // An Option is used to configure the profiler's behaviour.
 type Option func(*config)
 
+// WithAgentHost specified the hostname to use to reach datadog trace agent.
+func WithAgentHost(host string) Option {
```

Comment:
I suggest having just one option that takes “host:port”. Maybe this one as it is.